blobmodel/stochasticality.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from .blobs import Blob
+from nptyping import NDArray, Float
+from typing import Any
+
+
+class BlobFactory(ABC):
+    """
+    Abstract class used by 2d propagating blob model to specify blob parameters. 
+    """
+
+    @abstractmethod
+    def sample_blobs(
+        self, Ly: float, T: float, num_blobs: int, blob_shape: str, t_drain: float
+    ) -> list[Blob]:
+        raise NotImplementedError
+
+
+class DefaultBlobFactory(BlobFactory):
+    """
+    Defualt blob parameters 
```

Comment:
I would give a more detailed description here. Something like "Default implementation of \class{BlobFactory}. Generates blob parameters for different possible random distributions. All random variables are independent from each other"